tests/core_tests/validations/base_functions.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+
+import logging
+import time
+import asyncio
+from typing import Callable
+from typing import Any
+
+from everest.testing.core_utils.test_control_module import TestControlModule
+
+## helper functions
+
+## ---------------------------------------------------
+def get_key_if_exists(dataset:dict, keyset:list):
+    """This function traverses a dataset (json tree-like data structure) for one or more (consecutive) key(s).
+    It tries to match the keys form the 'keyset' list beginning from the dataset's root until it either locates
+    the last entry in the keyset (which it then returns) or until a key is not found in the dataset (thus returning 'None').
+
+    Args:
+        dataset (dict): Tree-like data structure to search in.
+        keyset (list): One or more consecutive keys to be found in the dataset.
+
+    Returns:
+        tuple(bool, Any): Key found (True/False), The contents of the found key or 'None'.
+    """
+    if not isinstance(dataset, dict):
+        return False, None
+    else:
+        if keyset is not []:
+            if len(keyset) == 1:
+                if keyset[0] in dataset:
+                    return True, dataset[keyset[0]]
+                else:
+                    return False, None
+            else:
+                return get_key_if_exists(dataset.get(keyset[0]), keyset[1:])
```

Comment:
have a look at **https://stackoverflow.com/questions/43491287/elegant-way-to-check-if-a-nested-key-exists-in-a-dict**, this is pretty close to what is needed here